lib/sidekiq_publisher/worker.rb

```diff
@@ -3,7 +3,9 @@
 module SidekiqPublisher
   module Worker
     def self.included(base)
-      base.include(Sidekiq::Worker)
+      # Prefer Sidekiq::Job over Sidekiq::Worker to support name change rolled
+      # out in sidekiq 7.0.0 per mperham/sidekiq#4971
+      base.include(defined?(Sidekiq::Job) ? Sidekiq::Job : Sidekiq::Worker)
```

Comment:
This is going to cause issues with [`sidekiq <= 6.2.1`](https://github.com/mperham/sidekiq/blob/50d0c3d2fdcab6eed57700bc5fac9a9e69fce7c0/lib/sidekiq/api.rb#L291). I also believe that `6.2.2` doesn't automatically require the job alias.

I'm not sure why the Appraisals didn't catch this, something else to look into, test-wise.

We should consider explicitly checking the version and acting appropriately instead.